src/mbgl/layout/pattern_layout.hpp

```diff
@@ -54,8 +75,12 @@ class PatternLayout : public Layout {
                 hasPattern = true;
             } else if (!constantPattern.to.empty()){
                 hasPattern = true;
+                auto patternPropertyValue = layer->as<PatternLayer>()->unevaluated.template get<typename PatternLayer::PatternProperty>().getValue();
```

Comment:
@kkaefer this feels messy and I'm open to suggestions about how to do this in a better way, but the flashing issue was caused by the fact that, by the time a feature-constant, zoom-varying pattern property gets to this step, it is already "constant" and the expression isn't available to evaluate with other zoom levels, so I needed to find a way to access the original `PropertyValue` so I could evaluate the expression with additional zoom levels to make sure we add all possible cross-fading dependencies.